src/tools/s3select.js

```diff
@@ -0,0 +1,50 @@
+/* Copyright (C) 2016 NooBaa */
+'use strict';
+
+const nb_native = require('../util/nb_native');
+const { Transform } = require('readable-stream');
+
+class S3SelectStream extends Transform {
+
+    constructor(opts = {}) {
+        super(opts);
+        const context = {
+            query: process.argv[2],
+            input_format: 'CSV',
+            input_serialization_format: {
+                FieldDelimiter: ',',
+                RecordDelimiter: '\n',
+                FileHeaderInfo: 'IGNORE'
+            },
+            records_header_buf: Buffer.from("don't care") /*Used for CRC, ignored here.*/
+        };
+        const nb_native_instance = nb_native();
```

Comment:
you can usually keep the nb_native function as is and just extract what you want to check from it like this:
```suggestion
        const { S3Select } = nb_native();
        if (!S3Select) throw new Error(...)
        this.s3select = new S3Select(...);
```